02-peer-protocol.md

```diff
@@ -1862,6 +1962,49 @@ it's simplest to only allow it to set fee levels; however, as the same
 fee rate applies to HTLC transactions, the receiving node must also
 care about the reasonableness of the fee.
 
+### Updating Block Height: `update_blockheight` (`option_will_fund`)
+
+An `update_blockheight` message is sent by initiator node.
+Like any update, it's first committed to the receiver's commitment
+transaction and then (once acknowledged) committed to the sender's. Unlike
+an HTLC, `update_blockheight` is never closed but simply replaced.
+
+No `update_blockheight` changes may be pending when sent (i.e. must be
+fully committed on both sides).
+
+1. type: 137 (`update_blockheight`)
+2. data:
+   * [`channel_id`:`channel_id`]
+   * [`u32`:`blockheight`]
+
+#### Requirements
+
+The node _responsible_ for initiating the channel:
+  - MUST send `update_blockheight` to ensure the CSV lock mandated by
+    `option_will_fund` will expire at the end of the funding lease.
+  - MUST send `update_blockheight` when their local blockheight updates
+
+The node _not responsible_ for initiating the channel:
+  - MUST NOT send `update_blockheight`.
+
+A receiving node:
+  - if the `update_blockheight` is less than the last received `blockheight`:
```

Comment:
"if the `lease_end - blockheight` is less than the receiver announced `to_self_delay` MUST fail the channel?" to avoid justice delays downgrads ?

Reply:
Hmm I think I'm missing something about your point here. Anywhere there's a to_self_delay, the csv is max(to_self_delay, lease_end - blockheight).

Reply:
Right, you're using OP_MAX directly in the script? So should be good with concern of justice delay downgrades, but still you can save few bytes of `witnessScript` by checking this during message reception or transaction construction ?

Reply:
The max is done prior to the construction of the script, the actual onchain script is unchanged.